File: python/examples/mongodb.py
from pymongo import MongoClient
from pshconfig import Config


def usage_example():

    # Create a new Config object to ease reading the Platform.sh environment variables.
    # You can alternatively use os.environ yourself.
    config = Config()

    # The 'database' relationship is generally the name of primary SQL database of an application.
    # It could be anything, though, as in the case here here where it's called "mongodb".
    credentials = config.credentials('mongodb')

    try:
        server  = '{0}://{1}:{2}@{3}:{4}/{5}'.format(
            credentials['scheme'],
            credentials['username'],
            credentials['password'],
            credentials['host'],
            credentials['port'],
            credentials['path']
        )

        client = MongoClient(server)

        collection = client.main.starwars

        post = {
            "name": "Rey",
            "occupation": "Jedi"
        }

        post_id = collection.insert_one(post).inserted_id

        document = collection.find_one(
            {"_id": post_id}
        )

        return 'Found {0} ({1})<br />'.format(document['name'], document['occupation'])

    except Exception as e:
        return e

File: golang/examples/mongodb.go
package examples

import (
  "fmt"
  "context"
  "time"
  psh "github.com/platformsh/config-reader-go/v2"
  "go.mongodb.org/mongo-driver/mongo"
  "go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func FormattedCredentialsMongoDB(creds psh.Credential) (string, error) {
  formatted := fmt.Sprintf("%s://%s:%s@%s:%d/%s",
    creds.Scheme, creds.Username, creds.Password, creds.Host, creds.Port, creds.Path)
  return formatted, nil
}

func UsageExampleMongoDB() string {

  // Create a NewRuntimeConfig object to ease reading the Platform.sh environment variables.
	// You can alternatively use os.Getenv() yourself.
  config, err := psh.NewRuntimeConfig()
  if err != nil {
    panic(err)
  }

  // Get the credentials to connect to the Solr service.
  credentials, err := config.Credentials("mongodb")
  checkErr(err)

  formatted, err := FormattedCredentialsMongoDB(credentials)
  if err != nil {
    panic(err)
  }

  ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
  client, err := mongo.Connect(ctx, options.Client().ApplyURI(formatted))
  checkErr(err)

  collection := client.Database("main").Collection("starwars")

  res, err := collection.InsertOne(ctx, bson.M{"name": "Rey", "occupation": "Jedi"})
  id := res.InsertedID

  cursor, err := collection.Find(context.Background(), bson.M{"_id": id})
  if err != nil {
    panic(err)
  }

  cursor.Next(context.Background())
  document := cursor.Current

  // var result struct {
  //   Value float64
  // }
  // filter := bson.M{"_id": id}
  // ctx, _ = context.WithTimeout(context.Background(), 5*time.Second)
  // err = collection.FindOne(ctx, filter).Decode(&result)
  // if err != nil {
  //     panic(err)
  // }

  // printf("Found %s (%s)<br />\n", $document->name, $document->occupation);


  return fmt.Sprintf("Found %s (%s)", document[0], document[1])


  return "Successfully connected!"
}
